### ISSUE TITLE
samples/net: the sample of zoap_server fails to add multicast address

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#2390]
**_Reported by gurucharan p:_**

while running the ZOAP server sample application, getting the  multicast address adding issue error because of which unable to proceed further with the current application.

Steps to replicate:
```
net-tools:
 	Terminal 1:
 	$ ./loop-socat.sh

 	Terminal 2:
 	$ sudo ./loop-slip-tap.sh

Zephyr:
 	$ cd samples/net/zoap_server
 	$ make pristine && make BOARD=qemu_x86 run   

Libcoap:
 	$ cd examples
 	$ sudo ./etsi_coaptest.sh -i tap0 2001:db8::1
```
QEMU output:
```
  AS      arch/x86/core/intstub.o
  AS      arch/x86/core/swap.o
  CC      arch/x86/core/thread.o
  CC      arch/x86/core/x86_mmu.o
  CC      arch/x86/core/fatal.o
  LD      arch/x86/core/built-in.o
  CC      arch/x86/soc/ia32/soc.o
  LD      arch/x86/soc/ia32/built-in.o
  LD      arch/x86/built-in.o
  LD      arch/built-in.o
  CC      src/zoap-server.o
  LD      src/built-in.o
  AR      libzephyr.a
  AR      libapplication.a
  LINK    zephyr.lnk
  SIDT    staticIdt.o
  GMMU    mmu_tables.o
  GDT     gdt.o
  BIN     zephyr.bin
To exit from QEMU enter: 'CTRL+a, x'
[QEMU] CPU: qemu32
qemu-system-i386: warning: Unknown firmware file in legacy mode: genroms/multiboot.bin

[zoap-server] [ERR] join_coap_multicast_group: Could not add multicast address to interface

[zoap-server] [ERR] main: Could not join CoAP multicast group
```

(Imported from Jira ZEP-2560)

**Comment by nashif** [comment]
**_by Jukka Rissanen:_**

https://github.com/zephyrproject-rtos/zephyr/pull/1225

**Comment by nashif** [comment]
**_by gurucharan p:_**

adding 
CONFIG_NET_IF_MCAST_IPV6_ADDR_COUNT=5, manually in the configuration file(prj.conf) seems to be resolving this issue. I have verified it on top of v1.9.0-rc2 release and its working fine after adding the above configuration line.

**Comment by nashif** [comment]
**_by Chandrakala Kempanna:_**

_gurucharan p_ Please Verify

**Comment by nashif** [comment]
**_by gurucharan p:_**

It seems this issue has been fixed with commit ID: eb5d6fcef5969b596dc0613c8b166bf55d9203b8

So closing the Jira.
